web-api/src/storage/postgres.rs

```diff
@@ -917,19 +927,17 @@ impl storage::Document for Storage {
                     Either::Right(document.id)
                 }
             });
-        let mut failed_documents = self.elastic.insert_documents(&candidates).await?;
-        failed_documents.extend(self.elastic.delete_documents(&noncandidates).await?);
-
+        let failed_documents = self.elastic.upsert_documents(&candidates).await?;
+        self.elastic.delete_by_parents(&noncandidates).await?;
```

Comment:
:art: newline before return
```suggestion
        self.elastic.delete_by_parents(&noncandidates).await?;

```